sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1003,4 +1007,94 @@ export class ApplicationService extends RecordDataModelService<Application> {
     application.coeStatus = COEStatus.declined;
     return this.repo.save(application);
   }
+
+  /**
+   * Associates an MSFAA number to the application checking
+   * whatever is needed to create a new MSFAA or use an
+   * existing one instead.
+   * @param applicationId application id to receive an MSFAA.
+   */
+  async associateMSFAANumber(applicationId: number): Promise<Application> {
+    const application = await this.repo.findOne(applicationId, {
+      relations: ["offering"],
+    });
+    if (!application) {
+      throw new CustomNamedError(
+        "Student Application not found.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.applicationStatus !== ApplicationStatus.assessment) {
+      throw new CustomNamedError(
+        `Student Application is not in the expected status. The application must be in application status '${ApplicationStatus.assessment}' to an MSFAA number be assigned.`,
+        INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+      );
+    }
+
+    let msfaaNumberId: number;
+
+    // Checks if there is an MSFAA that could be considered valid.
```

Comment:
'an MSFAA' , please change to 'an existing MSFAA '

Reply:
Please see the comments above, "an MSFAA" seems to be the correct way.